File: core/src/main/java/com/nike/redwiggler/core/FileBasedEndpointCallProvider.java
package com.nike.redwiggler.core;

import com.nike.redwiggler.core.models.EndpointCall;

import java.io.File;
import java.util.List;
import java.util.stream.Collectors;


public class FileBasedEndpointCallProvider implements EndpointCallProvider {
    private final List<File> files;

    public FileBasedEndpointCallProvider(List<File> files) {
        this.files = files;
    }

    @Override
    public List<EndpointCall> getCalls() {
        return files
                .stream()
                .map(EndpointCall::fromFile)
                .collect(Collectors.toList());
    }

}

File: core/src/main/scala/com/nike/redwiggler/core/FileBasedEndpointCallProvider.scala
package com.nike.redwiggler.core

import java.io.File

import com.nike.redwiggler.core.models.EndpointCall

case class FileBasedEndpointCallProvider(files : Seq[File]) extends EndpointCallProvider {

  def this(file: File) {
    this(Seq(file))
  }

  override def getCalls: Seq[EndpointCall] = files.map(EndpointCall.fromFile)
}
